leakcanary-android-core/src/main/java/leakcanary/DefaultOnHeapAnalyzedListener.kt

```diff
@@ -71,6 +89,44 @@ class DefaultOnHeapAnalyzedListener(private val application: Application) : OnHe
     )
   }
 
+  /**
+   * Android TV devices do not have notifications, therefore the only easy and non-invasive way
+   * to communicate with user is via Toast messages. These are used just to grab user attention and
+   * to direct them to Logcat where a much more detailed report will be printed.
+   */
+  private fun showToast() {
+    // Post the Toast into main thread and wrap it with try-catch in case Toast crashes (it happens)
+    handler.post {
+      try {
+        Toast.makeText(
+            application,
+            "Analysis complete, please check Logcat",
+            Toast.LENGTH_LONG
+        )
+            .show()
+      } catch (exception: Exception) {
+        // Toasts are prone to crashing, ignore
```

Comment:
What's this about? Toasts aren't prone to crashing, and if they were we should have specific code to handle those edge cases. The one crash that can happen is related to background / foreground, there are ways around it but catching will definitely do nothing as it's always crashing from a delayed message

Reply:
This comes from my old implementation of heap dump listener - I believe it was occasionally crashing after app was backgrounded - but that could've been because I was using Activity instead of Application for context. Will remove

Reply:
If you do repro the crash, open up an issue that way we'll track it